src/main/java/randoop/generation/OrienteeringSelection.java

```diff
@@ -0,0 +1,185 @@
+package randoop.generation;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.plumelib.util.CollectionsPlume;
+import randoop.sequence.ExecutableSequence;
+import randoop.sequence.Sequence;
+import randoop.util.Randomness;
+import randoop.util.SimpleList;
+
+/**
+ * Implements the Orienteering component, as described by the paper "GRT: Program-Analysis-Guided
+ * Random Testing" by Ma et. al (appears in ASE 2015):
+ * https://people.kth.se/~artho/papers/lei-ase2015.pdf .
+ *
+ * <p>Biases input selection towards sequences that have lower execution cost. Execution cost is
+ * measured by the number of method calls in a sequence and the time it takes to execute.
+ *
+ * <p>Our implementation of Orienteering differs from that described in the GRT paper in that we do
+ * not keep track of the execution time of an input sequence for each new run. Instead, we assume
+ * that an input sequence's execution time is equal to the execution time of its first run. We
+ * believe this assumption is reasonable since a sequence does not take any inputs (it is
+ * self-contained), so its execution time probably does not differ greatly between separate runs.
+ *
+ * <p>The GRT paper also does not describe how to handle input sequences that have an execution time
+ * of zero, such as one that only includes the assignment of a primitive type {@code byte byte0 =
+ * (byte)1;}. We assign these input sequences an execution time of 1 nanosecond to prevent division
+ * by zero when later computing weights.
+ */
+public class OrienteeringSelection extends InputSequenceSelector {
+  /** Map from a sequence to its weight. */
+  private final Map<Sequence, Double> weightMap = new HashMap<>();
```

Comment:
Are there any invariants among the 4 maps?  For example, do they all have the exact same set of keys?  Or might they have different keys?

Reply:
I see only two maps:  sequenceDetailsMap and weightMap.   I believe they have the same set of sequences as keys.